### PR TITLE
Feature: Qdrant filters supports

### PR DESCRIPTION
**Pull request by kacperlukawski**
# Support Qdrant filters

Qdrant has an [extensive filtering system](https://qdrant.tech/documentation/concepts/filtering/) with rich type support. This PR makes it possible to use the filters in Langchain by passing an additional param to both the `similarity_search_with_score` and `similarity_search` methods.

## Who can review?

@dev2049 @hwchase17

### CODE REVIEW

File: langchain/vectorstores/qdrant.py [hunk 1]
```diff
@@ -226,10 +227,21 @@ def similarity_search_with_score(
             List of Documents most similar to the query and score for each.
         """
 
+        if filter is not None and isinstance(filter, dict):
+            warnings.warn(
```
Comment (dev2049) [inline langchain/vectorstores/qdrant.py:231]: are we sure we want to warn here? if dicts are sufficient for some folks is there anything wrong with them keeping as is?

**Review by dev2049 — COMMENTED** [review]
nice! just one quick q

File: langchain/vectorstores/qdrant.py (hunk 1 above)
Comment (kacperlukawski) [reply langchain/vectorstores/qdrant.py:239]: @dev2049 The dict-based filtering covers some basic cases only and is not well-defined, to be honest. Recently there have been some changes introduced that made it even more complicated and I believe we should eventually get rid of them. I think there should be a single way exposed unless you have the plan to unify it for all the vector stores.

**Comment by gkourogiorgas** [comment]
This works great for me. What about .as_retriever()?
So far I am doing it manually by first searching for relevant documents using filters and then passing them as context manually to the prompt. It would be nice if I could do it in .as_retriever()

**Comment by kacperlukawski** [comment]
@gkourogiorgas This PR is pretty old and has already been merged. Could you please open an issue to discuss the feature you requested?

**Pull request merged**